mindmeld/active_learning/plot_manager.py

```diff
@@ -0,0 +1,474 @@
+        self.queries_data = self.get_queries_json_data()
+
+    # Get JSON Data
+    def get_accuracies_json_data(self) -> Dict:
+        """Loads accuracies.json from the experiment directory path.
+        Returns:
+            data (dict): Data loaded from accuracies.json.
+        """
+        accuracies_json_path = AL_ACCURACIES_PATH.format(
+            experiment_folder=self.experiment_dir_path
+        )
+        with open(accuracies_json_path, "r") as infile:
+            return json.load(infile)
+
+    def get_queries_json_data(self) -> Dict:
+        """Loads queries.json from the experiment directory path.
+        Returns:
+            data (dict): Data loaded from queries.json.
+        """
+        selected_queries_json_path = AL_SELECTED_QUERIES_PATH.format(
+            experiment_folder=self.experiment_dir_path
+        )
+        with open(selected_queries_json_path, "r") as infile:
+            return json.load(infile)
+
+    def queries_json_data_has_data(self) -> Dict:
+        """Checks whether queries.json is empty {}.
+        Returns:
+            has_data (bool): Whether queries.json has data.
+        """
+        return {} != self.get_queries_json_data()
+
+    # Plotting Meta Functions
+    def generate_plots(self):
+        """Entry point for generating plots as per the specifications provided in the config."""
+        logger.info("Starting to generate plots")
+        plot_functions = [
+            "plot_single_epoch",
+            "plot_avg_across_epochs",
+            "plot_all_epochs",
+        ]
+        for plot_function in plot_functions:
+            logger.info("Plotting: %s", plot_function)
+            plot_function = getattr(PlotManager, plot_function)
+            self.plotting_wrapper(plot_function)
+        if self.queries_json_data_has_data():
+            logger.info("Plotting: plot_stacked_bar")
+            self.plot_stacked_bar()
+
+    def plotting_wrapper(
+        self,
+        function,
+        plot_domain: bool = True,
+        plot_intents: bool = True,
+        plot_entities: bool = False,
+    ):
+        """Plotting wrapper functions for plots that use data from accuracies.json
+        Args:
+            function (generator): plotting function.
+            plot_domain (bool): Whether to generate plots at the domain level.
+            plot_intents (bool): Whether to generate plots at the intent level.
+            plot_entities (bool): Whether to generate plots at the entity level.
+        """
+        if plot_domain:
+            y_keys = ["overall"]
+            function(self, y_keys=y_keys)
+            for domain in self.get_domain_list():
+                y_keys = [domain, "overall"]
+                function(self, y_keys=y_keys)
+                if plot_intents:
+                    for intent in self.get_intent_list(domain):
+                        y_keys = [domain, intent, "overall"]
+                        function(self, y_keys=y_keys)
```

Comment:
`function(self, y_keys=[domain, intent, "overall"])`

Reply:
Should I do the same for the ones after as well?

```
    if plot_entities:
        y_keys = [domain, intent, "entities", "overall"]
        function(self, y_keys=y_keys)
        for entity in self.get_entity_list(domain, intent):
            y_keys = [domain, intent, "entities", entity]
            function(self, y_keys=y_keys)
```